cyclops-ctrl/internal/cerbos/db/users.go

```diff
@@ -0,0 +1,102 @@
+package db
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+type UserRecord struct {
+	Username string
+	Password string
+	Roles    []string
+}
+
+type UsersData struct {
+	Users []UserRecord `yaml:"users"`
+}
+
+type UserConfig struct {
+	clientset *kubernetes.Clientset
+}
+
+// NewUserConfig creates a new UserConfig with a Kubernetes clientset.
+func NewUserConfig() (*UserConfig, error) {
+	config := ctrl.GetConfigOrDie()
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	return &UserConfig{clientset: clientset}, nil
+}
```

Comment:
I would prefer to have a struct `UserDB` that would be initialized when the service starts and then create the clientset instead of constructing it each time we load a user config. You can also offload that to the K8s client since it already has a k8s clientset that can be queried for user secrets.